modules/zenith-public/lua/4-additive/quests/repeatableQuestDoubleRewards.lua

```diff
@@ -0,0 +1,664 @@
+        },
+
+        -----------------------------------
+        -- Jeuno
+        -----------------------------------
+        [243] = -- RULUDE_GARDENS
+        {
+            ['Taillegeas'] = { 3, 68 },  -- DUCAL_HOSPITALITY
+        },
+        [245] = -- LOWER_JEUNO
+        {
+            -- COMMUNITY_SERVICE (15) not implemented
+        },
+        [246] = -- PORT_JEUNO
+        {
+            -- ALL_IN_THE_CARDS (166) not implemented
+        },
+
+        -----------------------------------
+        -- Outlands / Other Areas
+        -----------------------------------
+        [247] = -- RABAO
+        {
+            ['Brave Ox'] = { 5, 14 },    -- A_DISCERNING_EYE (Outlands)
+            ['Agado-Pugado'] = { 5, 3 }, -- A_QUESTION_OF_TASTE
+        },
+        [248] = -- SELBINA
+        {
+            ['Falgima'] = { 4, 9 },      -- A_POTTERS_PREFERENCE
+            ['Melyon'] = { 4, 11 },      -- FISHERMANS_HEART
+            ['Valgeir'] = { 4, 18 },     -- ONLY_THE_BEST
+            ['Dortain'] = { 4, 7 },      -- ORLANDOS_ANTIQUES
+        },
+        [249] = -- MHAURA
+        {
+            ['Take'] = { 4, 19 },        -- AN_EXPLORERS_FOOTSTEPS
+            ['Rycharde'] = { 4, 20 },    -- CARGO
+        },
+        [250] = -- KAZHAM
+        {
+            ['Ronan'] = { 5, 1 },        -- THE_FIREBLOOM_TREE
+            ['Lalapp'] = { 5, 2 },       -- GREETINGS_TO_THE_GUARDIAN
+        },
+        [252] = -- NORG
+        {
+            ['Aeka'] = { 5, 130 },       -- BLACK_MARKET
+        },
+
+        -----------------------------------
+        -- Adoulin
+        -----------------------------------
+        [256] = -- WESTERN_ADOULIN
+        {
+            ['Gogosomi'] = { 9, 94 },    -- CAFETERIA
+            ['Epaunice'] = { 9, 34 },    -- FLOWER_POWER
+            ['Tibelda'] = { 9, 26 },     -- GRANDDADDY_DEAREST
+        },
+        [257] = -- EASTERN_ADOULIN
+        {
+            ['Mapolisa'] = { 9, 31 },    -- IT_NEVER_GOES_OUT_OF_STYLE
+        },
+    }
+
+    return npcQuestMapCache
+end
+
+-----------------------------------
+-- Helper: Check if quest is in whitelist
+-----------------------------------
+local function isRepeatableQuest(areaId, questId)
+    local quests = getRepeatableQuests()
+    local areaQuests = quests[areaId]
+
+    if areaQuests == nil then
+        return false
+    end
+
+    return areaQuests[questId] == true
+end
+
+-----------------------------------
+-- Helper: Get log ID from area parameter
+-----------------------------------
+local function getLogId(area)
+    if type(area) == 'number' then
+        return area
+    elseif area and area['quest_log'] then
+        return area['quest_log']
+    end
+
+    return nil
+end
+
+-----------------------------------
+-- Helper: Get daily completion count for a quest
+-----------------------------------
+local function getDailyCount(player, areaId, questId)
+    local varFullName = string.format('DailyGilBonus_%d_%d', areaId, questId)
```

Comment:
please move this string bulder into a helper. I don't like that the format of it is defined in so many places

Reply:
```suggestion
    local varFullName = getDailyVarName(areaId, questId)
```

and define the `local function` above

Reply:
fixed